tracer/src/Datadog.Trace.Tools.Runner/Checks/AgentConnectivityCheck.cs

```diff
@@ -19,20 +19,20 @@ namespace Datadog.Trace.Tools.Runner.Checks
 {
     internal class AgentConnectivityCheck
     {
-        public static Task<bool> Run(ProcessInfo process)
+        public static Task<bool> RunAsync(ProcessInfo process)
         {
             var settings = new ExporterSettings(process.Configuration);
 
             var url = settings.AgentUri.ToString();
 
             AnsiConsole.WriteLine(DetectedAgentUrlFormat(url));
 
-            return Run(new ImmutableExporterSettings(settings));
+            return RunAsync(new ImmutableExporterSettings(settings));
         }
 
-        public static async Task<bool> Run(ImmutableExporterSettings settings)
+        public static async Task<bool> RunAsync(ImmutableExporterSettings settings)
         {
-            var payload = Vendors.MessagePack.MessagePackSerializer.Serialize(Array.Empty<Span[]>());
+            var payload = Vendors.MessagePack.MessagePackSerializer.Serialize(Array.Empty<Span[][]>());
```

Comment:
I don't get this change

Reply:
The agent expects an array of arrays of spans (= an array of traces), not just an array of spans. It didn't cause any issue because they are serialized the same way (since the outer array is empty, the inner array isn't serialized at all, so `span[0]` produces the same output as `span[0][]`). It's really just a "code correctness" change.

Reply:
Cool thanks. We could do the same in the `AgentWriter` code to make it clearer then as well:
https://github.com/DataDog/dd-trace-dotnet/blob/master/tracer/src/Datadog.Trace/Agent/AgentWriter.cs#L57

Reply:
Good catch. I'll make that change in a separate PR

Reply:
ugh never mind, my change is actually wrong. `Array.Empty<span[]>` **does** produce a `span[][]`. `Array.Empty<span[][]>` produces a `span[][][]` 🤦‍♂️

Reply:
lol indeed. Sorry I should have caught it when you explained it :)

Reply:
Rubberduck FTW

Reply:
I think I mentioned this somewhere else, but an empty array will always be the byte `1001 0000` in MessagePack ([docs](https://github.com/msgpack/msgpack/blob/master/spec.md#array-format-family)), so if we wanted to we could reduce this to:
```
var payload = new byte[] { 144 };
```